File: QuizAppClient/src/ui/TimeThread.java
package ui;

import client.ClientMain;

import javax.swing.*;
import java.awt.*;
import java.sql.Time;

public class TimeThread implements  Runnable{
    private JTextField timeTxt;
    private  int time;
    private boolean isOver;
    private  JFrame parentFrame;
    TimeThread(JFrame parentFrame,JTextField timeTxt, int time){
        this.parentFrame = parentFrame;
        this.timeTxt = timeTxt;
        this.time = time;

        new Thread(this).run();
    }

    public boolean isOver() {
        return isOver;
    }

    @Override
    public void run() {

        for(int i=time; i>=0; i--){
            timeTxt.setFont(new Font(Font.SANS_SERIF,Font.BOLD,17));
            timeTxt.setText(i + "");
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
        isOver = true;
        ClientGui.disableAll();
        int score[] = ClientMain.sendResult();
        ClientGui.printScore(parentFrame,score);
    }
}

File: QuizAppClient/src/ui/TimeThread.kt
package ui

import client.ClientMain.sendResult
import java.awt.Font
import javax.swing.JFrame
import javax.swing.JTextField

class TimeThread internal constructor(private val parentFrame: JFrame, private val timeTxt: JTextField, private val time: Int) : Runnable {
    var isOver = false
        private set

    override fun run() {
        for (i in time downTo 0) {
            timeTxt.font = Font(Font.SANS_SERIF, Font.BOLD, 17)
            timeTxt.text = i.toString() + ""
            try {
                Thread.sleep(1000)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
        isOver = true
        ClientGui.Companion.disableAll()
        val score = sendResult()
        ClientGui.Companion.printScore(parentFrame, score)
    }

    init {
        Thread(this).run()
    }
}